packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -165,18 +165,69 @@ export default class ScriptTransformer {
 
     for (let i = 0; i < transformRegExp.length; i++) {
       if (transformRegExp[i][0].test(filename)) {
-        const transformPath = transformRegExp[i][1];
-        this._transformConfigCache.set(transformPath, transformRegExp[i][2]);
-
-        return transformPath;
+        return transformRegExp[i][1];
       }
     }
 
     return undefined;
   }
 
+  async loadTransformers(): Promise<void> {
+    await Promise.all(
+      this._config.transform.map(
+        async ([, transformPath, transformerConfig]) => {
+          let transformer: Transformer;
+
+          try {
+            transformer = require(transformPath);
```

Comment:
we might consider adding a `Map` to the module scope for the `require` and `import` calls. However, they should be by reference when called multiple times in the same context, so I don't think the overhead is necessarily worth it?